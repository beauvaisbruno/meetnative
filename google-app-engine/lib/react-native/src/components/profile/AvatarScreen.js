"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const react_navigation_1 = require("react-navigation");
const React = __importStar(require("react"));
const native_base_1 = require("native-base");
const Avatars_1 = require("./img/Avatars");
const react_native_1 = require("react-native");
const react_native_auto_height_image_1 = __importDefault(
  require("react-native-auto-height-image")
);
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
exports.AvatarScreen = ({ navigation }) => {
  const rows = [];
  let even = true;
  let cols = [];
  Object.keys(Avatars_1.avatars).forEach(index => {
    cols.push(
      React.createElement(
        react_native_1.TouchableHighlight,
        {
          onPress: () => {
            navigation.navigate("ProfileContainer", { avatar: index });
          },
          key: index,
          testID: index
        },
        React.createElement(react_native_auto_height_image_1.default, {
          source: Avatars_1.avatars[index],
          width: react_native_responsive_screen_1.widthPercentageToDP("50%")
        })
      )
    );
    if (!even) {
      rows.push(
        React.createElement(
          native_base_1.View,
          { key: index, style: { flexDirection: "row" } },
          cols
        )
      );
      cols = [];
    }
    even = !even;
  });
  return React.createElement(native_base_1.Content, null, rows);
};
exports.default = react_navigation_1.withNavigation(exports.AvatarScreen);
//# sourceMappingURL=AvatarScreen.js.map
