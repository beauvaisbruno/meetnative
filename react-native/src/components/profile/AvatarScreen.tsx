import { NavigationInjectedProps, withNavigation } from "react-navigation";
import * as React from "react";
import { Content, View } from "native-base";
import { avatars } from "./img/Avatars";
import { TouchableHighlight } from "react-native";
import AutoHeightImage from "react-native-auto-height-image";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

export const AvatarScreen = ({ navigation }: NavigationInjectedProps) => {
  const rows: JSX.Element[] = [];
  let even = true;
  let cols: JSX.Element[] = [];
  Object.keys(avatars).forEach(index => {
    cols.push(
      <TouchableHighlight
        onPress={() => {
          navigation.navigate("ProfileContainer", { avatar: index });
        }}
        key={index}
        testID={index}
      >
        <AutoHeightImage source={avatars[index] as number} width={wp("50%")} />
      </TouchableHighlight>
    );
    if (!even) {
      rows.push(
        <View key={index} style={{ flexDirection: "row" }}>
          {cols}
        </View>
      );
      cols = [];
    }
    even = !even;
  });

  return <Content>{rows}</Content>;
};

export default withNavigation(AvatarScreen);
