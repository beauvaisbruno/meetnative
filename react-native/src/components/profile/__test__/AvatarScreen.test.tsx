jest.mock("react-navigation", () => {
  return {
    withNavigation: (param: any) => param
  };
});

jest.mock("../img/Avatars", () => {
  return {
    avatars: ["avatar1", "avatar2"]
  };
});

import * as React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import { AvatarScreen } from "../AvatarScreen";

it("Age Selector", async () => {
  const navigate = jest.fn();
  // @ts-ignore
  const tree = shallow(<AvatarScreen navigation={{ navigate }} />);
  expect(toJson(tree)).toMatchSnapshot();
  const TouchableHighlight = tree.findWhere(
    node => node.prop("testID") === "1"
  );
  TouchableHighlight.props().onPress();
  expect(navigate).toBeCalledWith("ProfileContainer", { avatar: "1" });
});
