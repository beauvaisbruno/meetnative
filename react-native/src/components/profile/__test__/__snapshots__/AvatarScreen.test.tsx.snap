// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age Selector 1`] = `
<Styled(Content)>
  <Styled(ViewNB)
    key="1"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      key="0"
      onPress={[Function]}
      testID="0"
      underlayColor="black"
    >
      <ErrorableImage
        source="avatar1"
        width={375}
      />
    </TouchableHighlight>
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      key="1"
      onPress={[Function]}
      testID="1"
      underlayColor="black"
    >
      <ErrorableImage
        source="avatar2"
        width={375}
      />
    </TouchableHighlight>
  </Styled(ViewNB)>
</Styled(Content)>
`;
